omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/PatientAllergyController2_0Test.java

```diff
@@ -0,0 +1,244 @@
+		Assert.assertEquals("test non coded", Util.getByPath(newAllergy, "allergen/nonCodedAllergen"));
+		Assert.assertEquals("5622AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA", Util.getByPath(newAllergy, "allergen/codedAllergen/uuid"));
+		allergies = Context.getPatientService().getAllergies(patient);
+		Assert.assertEquals(allergies.size(), 5);
+	}
+
+	/**
+	 *  Set No Known Allergies icorrectly using empty post body
+	 *  achieved with PUT with empty post body when list of allergies is empty
+	 *  throw ResourceDoesNotSupportOperationException because Patient does not exist
+	 */
+	@Test(expected = ResourceDoesNotSupportOperationException.class)
+	public void shouldThrowExceptionWhenSettingNoKnownAllergiesForNonExistingPatientUsingEmptyPostBodyAllergiesNotEmpty() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Patient patient = allergy.getPatient();
+		Allergies allergies = Context.getPatientService().getAllergies(patient);		
+		Assert.assertEquals(Allergies.SEE_LIST, allergies.getAllergyStatus());
+
+		// atempt to set no known allergies with PUT
+		String nonExistentPatientURI = "patient/nonExistingPatient/allergy";
+		String json = "{}";
+		handle(newPutRequest(nonExistentPatientURI, json));
+	}
+	
+	/**
+	 *  Set No Known Allergies icorrectly using non empty post body and List of Allergies is empty
+	 *  achieved with PUT with empty post body when list of allergies is empty
+	 *  throw ResourceDoesNotSupportOperationException because of non empty post body
+	 */
+	@Test(expected = ResourceDoesNotSupportOperationException.class)
+	public void shouldThrowExceptionWhenSettingNoKnownAllergiesWithExistingPatientUsingNonEmptyPostBodyAllergiesEmpty() throws Exception
+	{
+		Allergy allergy = Context.getPatientService().getAllergyByUuid(getUuid());
+		Patient patient = allergy.getPatient();
+		Allergies allergies = Context.getPatientService().getAllergies(patient);		
+		Assert.assertEquals(Allergies.SEE_LIST, allergies.getAllergyStatus());
+		
+		// delete existing allergies
+		handle(newDeleteRequest(getURI(), 
+				new Parameter("reason", "unit test")));
+
+		allergies = Context.getPatientService().getAllergies(allergy.getPatient());
+		Assert.assertEquals(Allergies.UNKNOWN, allergies.getAllergyStatus());
+
+		// atempt to set no known allergies with PUT
+		String json = "{\"allergen\": {\"allergenType\":\"DRUG\", \"nonCodedAllergen\": \"test non coded\" }}";
+		handle(newPutRequest(getURI(), json));
+	}
+	
+	/**
+	 *  Set No Known Allergies icorrectly using empty post body
+	 *  achieved with PUT with empty post body when list of allergies is empty
+	 *  throw ResourceDoesNotSupportOperationException because List of Allergies is not empty
+	 */
+	@Test(expected = ResourceDoesNotSupportOperationException.class)
+	public void shouldThrowExceptionWhenSettingNoKnownAllergiesWithExistingPatientUsingEmptyPostBodyAllergiesNotEmpty() throws Exception

```

Comment:
Since the contract is that an empty post body sets the status to no known allergies, we can remove this from all unit test method names. And because this can only be set with existing patients, you can also remove the WithExistingPatient part.
In other words, you could simply say that: shouldThrowExceptionWhenSettingNoKnownAllergiesWhenAllergiesNotEmpty

Does this make sense?